bin/extract_galaxy_tools.py

```diff
@@ -516,9 +562,32 @@ def export_tools(
 
     if add_usage_stats:
         df = add_usage_stats_for_all_server(df)
+    
+   # df_edam = df[df['To keep']==True]
+    df_edam1 =df[df['EDAM operation'].notna()]
```

Comment:
Sorry but this seems far to complicated. Can you not just do
df[["EDAM operation (reduced)", "EDAM topic (reduced)"]] = df[["EDAM operation", "EDAM topic"]].map(reduced_edam_term)
and `reduced_edam_term` will take the EDAM term as input and return the reduced form.